command/pipe.go

```diff
@@ -0,0 +1,337 @@
+
+	dst, err := url.New(c.Args().Get(0), url.WithRaw(c.Bool("raw")))
+	if err != nil {
+		printError(fullCommand, c.Command.Name, err)
+		return nil, err
+	}
+
+	return &Pipe{
+		dst:          dst,
+		op:           c.Command.Name,
+		fullCommand:  fullCommand,
+		deleteSource: deleteSource,
+		// flags
+		noClobber:          c.Bool("no-clobber"),
+		storageClass:       storage.StorageClass(c.String("storage-class")),
+		concurrency:        c.Int("concurrency"),
+		partSize:           c.Int64("part-size") * megabytes,
+		encryptionMethod:   c.String("sse"),
+		encryptionKeyID:    c.String("sse-kms-key-id"),
+		acl:                c.String("acl"),
+		cacheControl:       c.String("cache-control"),
+		expires:            c.String("expires"),
+		contentType:        c.String("content-type"),
+		contentEncoding:    c.String("content-encoding"),
+		contentDisposition: c.String("content-disposition"),
+		dstRegion:          c.String("destination-region"),
+
+		// s3 options
+		storageOpts: NewStorageOpts(c),
+	}, nil
+}
+
+// Run starts copying stdin output to destination.
+func (c Pipe) Run(ctx context.Context) error {
+	if c.dst.IsBucket() || c.dst.IsPrefix() {
+		return fmt.Errorf("target %q must be an object", c.dst)
+	}
+	err := c.shouldOverride(ctx, c.dst)
+	if err != nil {
+		if errorpkg.IsWarning(err) {
+			printDebug(c.op, err, nil, c.dst)
+			return nil
+		}
+		return err
+	}
+	// override destination region if set
+	if c.dstRegion != "" {
+		c.storageOpts.SetRegion(c.dstRegion)
+	}
+	dstClient, err := storage.NewRemoteClient(ctx, c.dst, c.storageOpts)
+	if err != nil {
+		return err
+	}
+	metadata := storage.NewMetadata().
+		SetStorageClass(string(c.storageClass)).
+		SetSSE(c.encryptionMethod).
+		SetSSEKeyID(c.encryptionKeyID).
+		SetACL(c.acl).
+		SetCacheControl(c.cacheControl).
+		SetExpires(c.expires)
+
+	if c.contentType != "" {
+		metadata.SetContentType(c.contentType)
+	} else {
+		metadata.SetContentType(guessContentTypeByExtension(c.dst))
+	}
+	if c.contentEncoding != "" {
+		metadata.SetContentEncoding(c.contentEncoding)
+	}
+	if c.contentDisposition != "" {
+		metadata.SetContentDisposition(c.contentDisposition)
+	}
+	err = dstClient.Put(ctx, &stdin{file: os.Stdin}, c.dst, metadata, c.concurrency, c.partSize)
+	if err != nil {
+		return err
+	}
+	msg := log.InfoMessage{
+		Operation:   c.op,
+		Source:      nil,
+		Destination: c.dst,
+		Object: &storage.Object{
+			StorageClass: c.storageClass,
+		},
+	}
+	log.Info(msg)
+	if err != nil {
+		return &errorpkg.Error{
+			Op:  c.op,
+			Src: nil,
+			Dst: c.dst,
+			Err: err,
+		}
+	}
+	return nil
+}
+
+// shouldOverride function checks if the destination should be overridden if
+// the destination object and given pipe flags conform to the
+// override criteria.
+func (c Pipe) shouldOverride(ctx context.Context, dsturl *url.URL) error {
+	// if not asked to override, ignore.
+	if !c.noClobber {
+		return nil
+	}
+
+	dstClient, err := storage.NewClient(ctx, dsturl, c.storageOpts)
```

Comment:
there's only 1 client. you can use `client` as var name.

Reply:
still relevant. please ping my comments if you've any relevant change.

Reply:
Sorry, resolved now. https://github.com/peak/s5cmd/pull/587/commits/e128849a061b777e6e9eb233efabace52f53138b